recipes/_tomcat-attributes.rb

```diff
@@ -6,69 +6,92 @@
 node.default['artifacts']['catalina-jmx']['destination'] = "#{node['alfresco']['home']}/lib"
 node.default['artifacts']['catalina-jmx']['owner'] = node['alfresco']['user']
 
+# memcached specific jars
+# enable these jars IF the tomcat memcached_nodes attribute is set and is not empty
+using_memcached = !node['tomcat']['memcached_nodes'].empty?
+node.default['artifacts']['memcached-session-manager']['enabled'] = using_memcached
 node.default['artifacts']['memcached-session-manager']['groupId'] = 'de.javakaffee.msm'
 node.default['artifacts']['memcached-session-manager']['artifactId'] = 'memcached-session-manager'
 node.default['artifacts']['memcached-session-manager']['version'] = '1.9.3'
 node.default['artifacts']['memcached-session-manager']['type'] = 'jar'
 node.default['artifacts']['memcached-session-manager']['destination'] = "#{node['alfresco']['home']}/lib"
 node.default['artifacts']['memcached-session-manager']['owner'] = node['alfresco']['user']
 
+node.default['artifacts']['memcached-session-manager-tc7']['enabled'] = using_memcached
 node.default['artifacts']['memcached-session-manager-tc7']['groupId'] = 'de.javakaffee.msm'
 node.default['artifacts']['memcached-session-manager-tc7']['artifactId'] = 'memcached-session-manager-tc7'
 node.default['artifacts']['memcached-session-manager-tc7']['version'] = '1.9.3'
 node.default['artifacts']['memcached-session-manager-tc7']['type'] = 'jar'
 node.default['artifacts']['memcached-session-manager-tc7']['destination'] = "#{node['alfresco']['home']}/lib"
 node.default['artifacts']['memcached-session-manager-tc7']['owner'] = node['alfresco']['user']
 
+# enable spymemcached if the tomcat memcached_nodes attribute contains n1: and n2:
+needs_spymemcached = node['tomcat']['memcached_nodes'].include?('n1:') && node['tomcat']['memcached_nodes'].include?('n2:')
```

Comment:
If the cluster is still caching elsewhere then yes, but the check for n1: && n2: would fail. I can change it to check for n1: as a minimum rather than both.

A better way would be to have attributes on this e.g. ['tomcat']['memcached_nodes']['n1'] = .... and ['tomcat']['memcached_nodes']['n2'] = ... and then the template can iterate over them. If there's only one dont prepend with 'n1:' other prepend all. WDYT?